code/go/0chain.net/smartcontract/storagesc/kill.go

```diff
@@ -0,0 +1,179 @@
+package storagesc
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"0chain.net/smartcontract/stakepool/spenum"
+
+	cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/smartcontractinterface"
+	"0chain.net/chaincore/transaction"
+	"0chain.net/core/common"
+)
+
+type providerRequest struct {
+	ID string `json:"id"`
+}
+
+func (pr *providerRequest) decode(p []byte) error {
+	return json.Unmarshal(p, pr)
+}
+
+func (ssc *StorageSmartContract) killBlobber(
+	t *transaction.Transaction,
+	input []byte,
+	balances cstate.StateContextI,
+) (string, error) {
+	var err error
+	var conf *Config
+	if conf, err = ssc.getConfig(balances, true); err != nil {
+		return "", common.NewError("kill_blobber_failed",
+			"can't get config: "+err.Error())
+	}
+	if err := smartcontractinterface.AuthorizeWithOwner("kill_blobber_failed", func() bool {
+		return conf.OwnerId == t.ClientID
+	}); err != nil {
+		return "", err
+	}
+
+	var req providerRequest
+	if err := req.decode(input); err != nil {
+		return "", common.NewError("kill_blobber_failed", err.Error())
+	}
+
+	var blobber *StorageNode
+	if blobber, err = ssc.getBlobber(req.ID, balances); err != nil {
+		return "", common.NewError("kill_blobber_failed",
+			"can't get the blobber "+t.ClientID+": "+err.Error())
+	}
+	blobber.Kill()
+	if err = emitUpdateBlobber(blobber, balances); err != nil {
+		return "", common.NewError("kill_blobber_failed", err.Error())
+	}
+
+	if blobber.LastRewardPartition.valid() {
```

Comment:
can you please elaborate on this logic

Reply:
We want to killed partitions to not receive rewards. So we remove them from the partition containing blobbers to be rewarded.